src/structures/paging/mapper/mapped_page_table.rs

```diff
@@ -179,7 +179,9 @@ impl<'a, P: PageTableFrameMapping> Mapper<Size1GiB> for MappedPageTable<'a, P> {
         let flags = p3_entry.flags();
 
         if !flags.contains(PageTableFlags::PRESENT) {
-            return Err(UnmapError::PageNotMapped);
+            let cloned = p3_entry.clone();
```

Comment:
This clears the entry even if the huge page bit doesn't match which seems counter-intuitive to me given the traits/methods signature.

If we're going to use `unmap_ignore_present` to implement `unmap` (which I'm not yet sure we should/can), swapping the present and huge page checks would be a small breaking change. If not, swapping the checks seems like a reasonable solution to me.

Reply:
I moved the check for huge pages before the present check. So this should be fixed now.